pkgs/applications/misc/j4-dmenu-desktop/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, cmake, dmenu }:
+
+stdenv.mkDerivation rec {
+  name    = "j4-dmenu-desktop-${version}";
+  version = "2.14";
+
+  src = fetchFromGitHub {
+    owner  = "enkore";
+    repo   = "j4-dmenu-desktop";
+    rev    = "r${version}";
+    sha256 = "14srrkz4qx8qplgrrjv38ri4pnkxaxaq6jy89j13xhna492bq128";
+  };
+
+  patchPhase = ''
+    substituteInPlace ./src/Main.hh \
+      --replace "\"dmenu -i\"" "\"${dmenu}/bin/dmenu -i\""
+  '';
+
+  buildInputs = [ cmake dmenu ];
```

Comment:
(In case you were unaware, adding stuff to inputs only directly affects the environment that the builder runs in; the run-time dependencies are calculated by other means).